src/extensions/askUserForConsent.ts

```diff
@@ -0,0 +1,100 @@
+import * as _ from "lodash";
+import * as clc from "cli-color";
+import * as marked from "marked";
+import TerminalRenderer = require("marked-terminal");
+
+import { FirebaseError } from "../error";
+import { logPrefix } from "../extensions/extensionsHelper";
+import * as iam from "../gcp/iam";
+import { promptOnce, Question } from "../prompt";
+import * as utils from "../utils";
+
+marked.setOptions({
+  renderer: new TerminalRenderer(),
+});
+
+/**
+ * Returns a string that will be displayed in the prompt to user.
+ * @param {string} role
+ * @return {!Promise<string>}
+ */
+export async function formatDescription(extensionName: string, projectId: string, roles: string[]) {
+  const question = `${clc.bold(
+    extensionName
+  )} will be granted the following access to project ${clc.bold(projectId)}`;
+  const results: string[] = await Promise.all(
+    roles.map(async (role: string) => {
+      return await retrieveRoleInfo(role);
+    })
+  );
+  results.unshift(question);
+  return _.join(results, "\n");
+}
+
+/**
+ * Returns a string representing a Role, see
+ * https://cloud.google.com/iam/reference/rest/v1/organizations.roles#Role
+ * for more details on parameters of a Role.
+ * @param {string} role
+ * @return {!Promise<string>}
+ */
+export async function retrieveRoleInfo(role: string) {
+  const res = await iam.getRole(role);
+  return `- ${res.title} (${res.description})`;
+}
+
+/**
+ * Displays roles and corresponding descriptions and asks user for consent
+ * @param {Array<string>} roles
+ * @return {Promise<?>}
+ */
+export async function prompt(extensionName: string, projectId: string, roles: string[]) {
+  if (!roles || !roles.length) {
+    return Promise.resolve();
```

Comment:
Returning a resolved promise is unnecessary in an `async` function.

```suggestion
    return;
```